Allow Credstash lookup plugin to return a default

gibbsoft opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
This PR Fixes #32346 and enables the credstash plugin to return a default
value when the lookup does not find the requested key in the credstash
table.

Features:

- Should `default` be undefined, we should preserve existing behaviour
- Should `default` be defined, we should return it upon `ItemNotFound`
- Update credstash plugin docs to demonstrate `default` parameter

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
credstash plugin

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.2.0.0
```
However, all current versions suffer this same issue.

##### ADDITIONAL INFORMATION
For more information, please see [Issue #32346](https://github.com/ansible/ansible/issues/32346)

evgeni committed:
format error messages from module_utils.basic nicer

* lists are printed as a string, separated by a comma and a space
* require_if tells if is_one_of was True or False

abadger committed:
Fix typo

abadger committed:
Update tests for required_if changes

These tests are doing string matches on the error condition.  Update
them to match the new strings.  This is probably okay to push out to old
releases even though it's technically backwards incompatible because
production playbooks won't be checking that a parameter was missing.
Param missing is something detected and fixed while writing the playbook.

bcoca committed:
remove hosts from removed when rescuing

fixes #31707

bcoca committed:
move to use ansible logging

bcoca committed:
fix closure issues

fixes #31786

jctanner committed:
Allow the user to declare an env var to set the vcsim container image

jctanner committed:
Enable tracing and log collection in the vcsim container

mattclay committed:
Handle bad options syntax in validate-modules.

bcoca committed:
fix html formatting

jcortejoso committed:
Define new_target_group before. Fixes #31378 (#31379)

lmprice committed:
Update the NetApp team (#31794)

Add a new maintainer to the NetApp team.

bcoca committed:
py2/py3 safer shas on hostvars (#31788)

* py2/py3 safer shas on hostvars

* plain bytes

chrismeyersfsu committed:
tests for InventoryModule error conditions (#31381)

* tests for InventoryModule error conditions

* modified unicode in tests to ahear to Ansible best practices

* flake8 fixes

jborean93 committed:
win_dsc: link module documentation to main RST page (#31814)

kedarkekan committed:
Update BOTMETA.yml

jctanner committed:
Fallback to the newer object type for dvswitch

jctanner committed:
Remove duplicate import

bcoca committed:
correctly deal with changed (#31812)

jborean93 committed:
win_regedit: fix extra info coming into stdout (#31813)

jborean93 committed:
fixed up code-block for powershell block

nazo committed:
skip check environment parameter if has no containers. (#31083)

* skip check environment parameter if has no containers.

* optimize code

prasadkatti committed:
Support 'termination protection' for cloudformation stacks (#31675)

* Support 'termination protection' for cloudformation stacks

- Pass in the stack_name and desired termination protection state to update_termination_protection

* Fix for failing cloudformation unit test

* Check if cfn has update_termination_protection attr

* Use hasattr to test if cfn supports update_termination_protection

* termination_protection shouldn't prevent update_stack call for existing stacks

paulquack committed:
Network command module for Brocade IronWare routers (#31429)

adambanker committed:
Fixed redshift_subnet_group typo (#31836)

eikef committed:
zabbix: Move commonly used docs to doc fragments (#31833)

* Move commonly used configuration parameter documentation for zabbix_*-modules to cental doc fragment

* fix indentation

eikef committed:
add validate_certs to zabbix_template and move documentation for common options to zabbix module_doc_fragment (#31838)

mattclay committed:
Fix bad syntax in test_infra integration test.

pilou- committed:
Reenable 'test' label for test related pull requests (#31472)

* Reenable 'test' label

See #29660 comments

* packaging is not a valid GitHub label anymore

samdoran committed:
Ensure valid_plugin_bin is defined

Fixes #31824

mkrizek committed:
hostname: fix for a new version of rhel (#31839)

Fixes #31811

fennm committed:
ec2_vpc_endpoint: make policy_path usable (#31801)

The current code flow precludes the use of the policy_path module
parameter that's documented.  It's actually called policy_file in the
code.

What's worse is that the policy_file branch actually tries to open the
file named by the policy parameter, even though policy and policy_file
are marked as mutually-exclusive.

This change fixes the logic bug in policy_file and updates the
documentation to reference policy_file.  The old parameter policy_path
is provided as an alias

willthames committed:
Allow AWS region to be detected from boto profiles (#31768)

mattclay committed:
Improve python 2/3 ABC fallback for pylint. (#31848)

* Improve python 2/3 ABC fallback for pylint.
* Allow longer method names in ansible-test.

jborean93 committed:
win_regedit: added function to load a dat file for editing (#31289)

* win_regedit: added function to load a dat file for editing

* try to make the tests more resiliant

* more stability changes

mattclay committed:
Disable flakey win_rabbitmq_plugin test.

weisslj committed:
[cloud] Add CAA record type to route53 (#29178)

The CAA record type is supported in Amazon Route 53 since Aug 21, 2017. Tested manually.

- https://aws.amazon.com/about-aws/whats-new/2017/08/amazon-route-53-now-supports-caa-records/
- http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/ResourceRecordTypes.html#CAAFormat
- https://tools.ietf.org/html/rfc6844

zanglang committed:
Remove extraneous comment block (#31865)

guilhermeoki committed:
Add support to pass a Makefile (#31853)

gundalow committed:
ovirt, azure_rm_common (#31877)

machacekondra committed:
ovirt: Add env variables to dynamic inventory (#31821)

This patch add new environment variables to oVirt dynamic inventory
to be consistent with all other oVirt modules:

 OVIRT_URL
 OVIRT_USERNAME
 OVIRT_CAFILE
 OVIRT_PASSWORD

Those variables are used as fallback if user don't specify a ini file,
with appropriate variables there.

zanglang committed:
Fix exception upon display.warn() (#31876)

Fixes #31875

mkrizek committed:
Add yum integration tests using fake repo (#31646)

jctanner committed:
Bump the vcsim container version (#31861)

gundalow committed:
Document backup_path (#31844)

Fix a few consistency issues at the same time

s-hertel committed:
[cloud] Check instance public-IP property before transforming to dict (#31849)

nomarek committed:
ec2_group: do not fail on description mismatch (#31704) (#31734)

* ec2_group: do not fail on description mismatch (#31704)

* ec2_group: do not fail on description mismatch (#31704) - fix test case

bcoca committed:
add all group vars back to implicit localhost

fixes #31857
reverts #31425

dagwieers committed:
Fix small typo

tacatac committed:
Add nosh service manager module (#31847)

* Add nosh service manager module

* based on the `svc`, `systemd`, `runit` and proposed `rc_service`
modules
* uses the high-level 'system-control' command and assumes nosh-native
interfaces though it should work with daemontools-style service scanning
* assumes a single service name is provided

* Metadata fixes

* Added "author" and "version_added"
* fixed the RETURN yaml

* PEP8 fixes

* fixed spacing issue

mattclay committed:
Fix docker_secret integration test dependencies.

caphrim007 committed:
Adds new module allowing you to wait for a bigip (#31846)

Module allows you to wait for a bigip device to be
"ready" for configuration. This module will wait for things like
the device coming online as well as the REST API and MCPD being
ready.

If all of the above is not online and ready, then no configuration
will be able to be made.

s-hertel committed:
Fix header direct_connect module_utils

alikins committed:
Relicense fact code I wrote to 2 clause BSD

abadger committed:
Relicense convert_bool to 2-clause BSD

This code originated in module_utils/basic.py which was BSD licensed.
In moving it and making it aplicable to other pieces of code that were
using similar functions, I added onto it a little.

nitzmahone committed:
Add Simplified BSD license to repo (#31913)

* changed basic.py Simplified BSD license header to new 1-line format
* created `licenses` dir at repo root to contain any future secondary licenses

s-hertel committed:
iam_managed_policy: use python 3 compatible policy comparison - fixes #31474 (#31535)

* Move compare_policies and hashable_policy functions into module_utils/ec2

* Use compare_policies which is compatible with python 2 and 3.

* rename function to indicate internal use

* s3_bucket: don't set changed to false if it has had the chance to be changed to true already.

dixhuit committed:
Typo on "Prompts" page (#31882)

mattclay committed:
Remove unsupported comments from aliases.

bcoca committed:
fixed caches (#30667)

tedder committed:
add root_device_type filter. (#31911)

* add root_device_type filter.

This module is due for a refactor, but I'm piling on.

* add version added

mkrizek committed:
git: remove duplicate ssh_wrapper cleanup (#31827)

mikedlr committed:
Mdd module unit test docs (#31373)

* new documentation for unit testing - especially module unit testing

* unit test documentation reformatting and further fixes

* unit test documentation - point to online coverage reports & fix bad spaces

* Small copy edits.

* First pass copy edit / rewrite. More info needed.

* testing documentation - clean up structure, especially code coverage - reduce repetition

* module unit test documentation - improved introduction

* testing documentation - more fixes from and inspired by review from dharmabumstead

* testing documentation - fixes from mattclay + some other minor tweaks

* More copy edits.

* testing documentation - further fixes from review

* Copy edits

* Copy edits

* More copy edits.

mattclay committed:
Add new default Docker container for ansible-test. (#31944)

* Add new default Docker container for ansible-test.
* Update ansible-test change classification.
* Update list of disabled pylint rules.
* Fix pylint issues with ansible-test.

maxamillion committed:
firewalld: don't reference undefined variable in error case (#31949)

* firewalld: don't reference undefined variable in error case

Signed-off-by: Adam Miller <maxamillion@fedoraproject.org>

* firewalld: don't set exception as var and not use it

Signed-off-by: Adam Miller <maxamillion@fedoraproject.org>

dagwieers committed:
Add UCS maintainers

Also fix to use implicit labeling (based on path) and create an aci_team entry

jmighion committed:
Adds regex for stdout needed for changing root password and regex for stderr catching problems with the password. (#31955)

bcoca committed:
dont add all group vars to implicit on create

they already get added in vars manager on use.
fixes #31420

nwsparks committed:
new windows module, win_audit_rule (#30473)

* added win_audit_rule with integration test

* Updated integration testing to target files as well as directories
and registry keys. Split testing files apart to be more organized.

Updated powershell for better handling when targetting file objects
and optimized a bit. Removed duplicated sections that got there from a
previous merge I think.

* Decided to make all the fact names the same in integration testing.
Seemed like there would be less change of accidentally using the wrong
variable when copy/pasting that way, and not much upside to having
unique names.

Did final cleanup and fixed a few errors in the integration testing.

* Fixed a bug where results was displaying a wrong value

Fixed a bug where removal was failing if multiple rules existed due to
inheritance from higher level objects.

* Resolved issue with unhandled error when used didn't have permissions
for get-acl.

Changed from setauditrule to addauditrule, see comment in script for reasoning.

Fixed state absent to be able to remove multiple entries if they exist.

* fixed docs issue

* updated to fail if invalid inheritance_rule when defining a file rather than warn

akomic committed:
New module: Add monitoring/zabbix_proxy module. (#24657)

* Implements zabbix_proxy module.

* Fixes formatting

* Bugfix

* Bugfix and various module documentation fixes.

* Iteritems python3 compatibility fix.

* PEP8 cleanup

Commit:
minor text fixes

willthames committed:
Handle timezone updates on Ubuntu 16.04+ on containers (#27546)

* Handle timezone updates on Ubuntu 16.04+ on containers

Although Ubuntu 16.04 will use timedatectl by default,
containers without a working timedatectl need to use the
old method.

A bug in Ubuntu for the old method means having to write
a nasty hack

https://bugs.launchpad.net/ubuntu/+source/tzdata/+bug/1554806

* Add tests for timezones

Ensure timezone changes work across various OSs

AmbientLighter committed:
Fix typo in ipa_user module documentation example. (#31928)

argo1984 committed:
Added Support for creating a Zabbix Inventory. It is possible to tag the (#23792)

* Rebase with update of remote repository

* Add Example

* Reference to example

* Fix error with collon (ansibot saw a yaml not a string)

* Change inventory mode to manual
add link to inventory documentation of zabbix

* Fix:
The test ansible-test sanity --test pep8 [?] failed with the following error:

lib/ansible/modules/monitoring/zabbix_host.py:532:1: E302 expected 2 blank lines, found 1
The test ansible-test sanity --test validate-modules [?] failed with the following error:

lib/ansible/modules/monitoring/zabbix_host.py:0:0: E309 version_added for new option (inventory_zabbix) should be 2.5. Currently 2.4

machacekondra committed:
ovirt_host_networks: Fix label assignment (#31969)

machacekondra committed:
ovirt_vms: Don't start VM if state is present (#31698)

gdcrocx committed:
Updated win_owner.py documentation to reflect recurse options (#31974)

Updated win_owner.py documentation to reflect recurse options 'yes' and 'no' instead of 'True' and 'False'

eikef committed:
Fix description parameter to check_all_properties (which was mangled in an earlier update) (#31987)

bcoca committed:
remove fixme from docs

eikef committed:
zabbix_proxy: implement validate_certs and use mod_doc_fragment (#31990)

* use zabbix mod_doc_fragment and implement validate_certs

* Fixes example to match documentation and provided choices

mattclay committed:
Fix ansible-test default image. (#31966)

* Add openssh-client to default docker container.
* Include Azure requirements in default container.

To do so, handling of pip requirements was updated to install each
set of requirements separately and then run a verification pass to
make sure there are no conflicts between requirements.

* Add missing --docker-no-pull option.
* Add documentation for the azure-requirements test.

bcoca committed:
make sure patterns are strings

fixes #31978

bcoca committed:
fix inventory loading for ansible-doc

fixes #31995

mattclay committed:
Add missing packages to default docker image.

nwsparks committed:
add newly merged windows module (#31989)

per: https://github.com/ansible/ansible/pull/30473

caphrim007 committed:
Refactors the bigip_gtm_facts module (#31917)

Includes pep fixes and inlining code with current conventions

caphrim007 committed:
Adds the bigip_policy module (#31915)

this module allows one to manage policies and re-order their
corresponding rules

eikef committed:
move zabbix modules into their own namespace below monitoring/ (#31992)

resmo committed:
composer: return skipped if command has no --dry-run (#32010)

keslerm committed:
Remove toLower on source (#31983)

Having this here breaks any source URLs that require case sensitivity.

jborean93 committed:
windows argv to string module utility (#28970)

* windows argv to string module utility

* rebased PR with latest devel branch

jborean93 committed:
win_command win_shell: add stdin option (#31619)

trishnaguha committed:
fix nxos_igmp_snooping (#31688)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

jmighion committed:
Using shorthand for disabling paging to avoid the syntax difference between versions. (#31942)

Fixes #31855

yaacov committed:
Enable use of empty manageiq arguments (#31774)

* anable use of empty manageiq arguments

* make default verify_ssl true

kedarkekan committed:
-Fixes JSON parsing(use JSON object instead of string) for facts modules. (#31818)

tacatac committed:
nosh system module: fixes and improvements (#32014)

* nosh system module: fixes and improvements

documentation:
* fleshed out and fixed to better follow the official guidelines

consistency:
* the following facts will now always be returned on success: name,
service_path, enabled, preset, user, status
* state is only returned when the state option is used
* state and status will be null if the service is not loaded by the end
of the task

* [nosh]: PEP8 fix

charlestolley committed:
Added regex to recognize prompt (#31947)

mkrizek committed:
ini inventory: document value parsing workaround

Fixes #31656

s-hertel committed:
Remove elasticdog from pacman module maintainers. (#32051)

ryansb committed:
Start using ClientRequestTokens in event lists (#31997)

* Start using ClientRequestTokens in event lists

* Include request token in all reqs that support it (basically all but check mode/changeset)

* Update placebo recordings

* Add comments for CRQ popping

nitzmahone committed:
fix #Requires to be valid in real Powershell (#31858)

* module_utils #Requires should not have .psm1 extension if "real" Powershell will ever execute them
* updated validate-modules to enforce this
* added check to disallow multi-module syntax on Ansible.ModuleUtils #Requires

abadger committed:
Prefer the stdlib SSLContext over urllib3 context

We do not go through the effort of finding the right PROTOCOL setting if
we have SSLContext in the stdlib.  So we do not want to hit the code
that uses PROTOCOL to set the urllib3-provided ssl context when
SSLContext is available.  Also, the urllib3 implementation appears to
have a bug in some recent versions.  Preferring the stdlib version will
work around that for those with Python-2.7.9+ as well.

Fixes #26235
Fixes #25402
Fixes #31998

jborean93 committed:
windows: add #AnsibleRequires for Windows modules (#31683)

* windows: add #AnsibleRequires to set whether a module requires module or a specific version

* fix up pep8 issues

* changed psversion to use the actual ps Requires -Version syntax

* missed the check on #Requires -Version

* fix #Requires module extensions

privateip committed:
fix eos terminal plugin to recognize ospf error message (#32039)

The eos terminal plugin did not correctly catch the error message
returned with trying to configure more than one ospf instance.  This
change updates the terminal plugin to catch that scenario

xscript committed:
add user-agent to Azure API calls (#31872)

* addi Ansible user-agent in Azure API calls

* fix import error

kedarkekan committed:
- Adds iosxr_netconf module to configure netconf service on IOSXR (#31715)

* - Adds iosxr_netconf module to configure netcong service on Cisco
  IOS-XR devices

* - Adds Integration test for module
- Handles diff return from load_config

* - Adds unit test for iosxr_netconf module

ogenstad committed:
Fix documentation bug for asa_config (#32007)

The documentation mentioned the "commit" and "update" parameters which
didn't exist.

gundalow committed:
Network docs should link to network integration pages (#31795)

gundalow committed:
Correct formatting --arguments (#31808)

* Correct formatting

* Use RST :option:

* ansible-pull --vault-password-file

* Streamlined the language a bit in the intro.

* Exclamation point removal!

mkrizek committed:
yum: allow for downgrade using rpm file (#31647)

bcoca committed:
removing deepcopy as it does not work

it cannot  handle deep crossreferenced objects
need to find alternative approach.

bcoca committed:
now with_dict allows for direct 'lookup' usage

fixes #32067

akatch committed:
Fun with callbacks (#32031)

* Functional-ish unix-ish callback plugin

* fix var

* Don't display include or include_vars

* Correctly display debug output and failed output in loops

* Linter fixes, documentation, and GPL oneliner

alikins committed:
Fix service_mgr fact collection (#32086)

The platform/distro/etc facts were being passed in
correctly, but service_mgr.py was looking up the
wrong names ('system' vs 'ansible_system') resulting
in service_mgr falling back to default 'service' result.

Fixes #30753, #31095

prasadkatti committed:
[ec2-ami] Use required_if with state parameter (#32009)

* [ec2-ami] Use required_if with state parameter

* Update failure msg on delete ami with no image_id task in integration test

Commit:
Add support for tags and instance attributes for destination variable for the ec2 dynamic inventory. (#29033)

samdoran committed:
Add ansible_distribution_major_version to macOS (#31708)

caphrim007 committed:
Segment out the update method (#32103)

This patch changes the base parameters class to segment out the
update method. This is done often throughout the f5 modules, so
this just bakes it into the mod utils so that I can delete it
from each module

jborean93 committed:
added .vscode to gitignore file (#32106)

s-hertel committed:
ec2_group: add rule description support - fixes #29040 (#30273)

* ec2_group: add support for rule descriptions.

* Document rule description feature and add an example using it.

* Fix removing rule descriptions.

* Add integration tests to verify adding/modifying/removing rule descriptions works as expected.

* Add permissions to hacking/aws_config/testing_policies/ec2-policy.json for updating ingress and egress rule descriptions.

* ec2_group: add backwards compatibility with older versions of botocore for rule descriptions.

* Add compatibility with older version of botocore for ec2_group integration tests.

* ec2_group: move HAS_RULE_DESCRIPTION to be checked first.

* Make requested change

* Pass around a variable instead of client

* Make sure has_rule_description defaults to None

* Fail if rule_desc is in any ingress/egress rules and the the botocore version < 1.7.2

* Remove unnecessary variable

* Fix indentation for changed=True when updating rule descriptions.

* minor refactor to remove duplicate code

* add missing parameter

* Fix pep8

* Update test policy.

privateip committed:
inserts enable cmd hash with auth_pass used (#32107)

eapi transport was not passing the auth_pass to the remote device with
it was provided.  this fix will now insert the correct command hash into
the jsonrpc request.

fixes #30802

rcarrillocruz committed:
Replace GigabitEthernet0/0/0/5 for GigabitEthernet0/0/0/2 in iosxr_interface intent tests (#32116)

There's no 5 interface present in CI nodes.

rcarrillocruz committed:
Add ansible_ssh_port to iosxr_user auth tests (#32117)

The CI nodes listen on port 8022, we need to plumb that to avoid
test failures.

rcarrillocruz committed:
Default ansible_ssh_port to 22 on ios_user tests (#32118)

The csr nodes do not have ansible_ssh_port on group_vars, as it uses
standard 22 port, therefore default to it on tests needing it.

rcarrillocruz committed:
Replace Gigabit0/0/0/2 for Gigabit0/0/0/1 on iosxr_interface.intent (#32120)

rcarrillocruz committed:
Introspect platform before running ios tests (#32125)

In some IOS tests, we can't just assume a certain interface, as
we test on IOS and IOS-XE and in our CI they scheme numbering is
different.
Introspect version on the tests where needed.

rcarrillocruz committed:
Set authorize yes on IOS interface discovery tests (#32128)

rcarrillocruz committed:
Fix quotes on ios_ping test (#32131)

gundalow committed:
consul_kv: Fix name & formatting (#32132)

* Fix name & formatting
* Fix examples

otteydw committed:
Add a validate example to blockinfile. (#32088)

pmarques committed:
[cloud] Add new module digital_ocean_sshkey_facts (#26365)

* do_sshkeys_facts module

* Fix

* Fix version

* Configure timeout and validate_certs for fetch_url

* Add support for new DO_OAUTH_TOKEN env var

* rename module prefix from do_* to digital_ocean_*

* Fix python 2.6 positional index

* Use modules_utils digital ocean lib

* Update metadata version

* Update version added

* Update module from boilerplate

* Update examples

* Add timeout documentation

* Fix typo

ryansb committed:
Add DO SSH key facts module to changelog

rcarrillocruz committed:
Discover ios_interface interface on basic tests (#32149)

We run these tests on IOS and IOS-XE, which have different interfaces
available for performing tests.

rahushen committed:
NXOS: Integration tests to Ansible (part 3) (#29030)

* Add nxos_file_copy IT

* Restructure nxos_igmp tests

* add nxos_igmp_interface IT

* add nxos_igmp_snooping IT

* add nxos_ntp_auth IT

* Add nxos_ntp_options IT

* update nxos.yaml with new tests

* update nxos_ntp_options test

* update nxos_ntp_auth IT

rcarrillocruz committed:
Set speed to 1000 on ios_interface tests (#32153)

csr and ios have only 1000 as value in common, using 10 or 100 will
cause CI failures.

shepdelacreme committed:
[cloud] Update ec2_vpc_net_facts to use Boto3 (#25375)

* update ec2_vpc_net_facts module to boto3

updated with RETURN values and other requested changes

removed errant extra blank line

another errant extra line removed...auto-linter not working apparently

updates per review

fix typo in RETURN docs

* fix trailing whitespace issue

abadger committed:
Update 2.4.1 release date

abadger committed:
Update release and maint for new release

caphrim007 committed:
Various bigip_pool fixes (#32161)

* corrects copyrights and mocks in unit tests
* fixes module code to include code to cleanup tokens

huslage committed:
Change 'connection' to 'inventory' in description

Fixing a copy/paste typo.

abadger committed:
reverse order of release numbers

caphrim007 committed:
Adds name fact when retrieving nodes (#32165)

wildmichael committed:
Add delayed start to win_nssm (#32017)

* Adds start_mode=delayed option for win_nssm

It translates to Start=SERVICE_DELAYED_AUTO_START in nssm.

* Adds documentation for the start_mode=delayed option of win_nssm

Also, reformats the start_mode values into a list.

sdodsley committed:
Fix issue #32138 - fix examples typo (#32140)

jmighion committed:
Adding a regex for stderr catching. (#31945)

bcoca committed:
Avoid default inventory proccessing for pull (#32135)

* Avoid default inventory proccessing for pull

- now pull's own special inventory processing should work correctly
- also removed ineffective set_defaults

fixes #31449

* use class property instead

* only do localhost for adhoc

(cherry picked from commit aad5d1432583c4aa4105b774f38c80498e85de59)

Commit:
Fix ec2_lc failing to create multi-volume configurations (#32163)

dathelen committed:
eos_user: sends user secret first on user creation fixes #31680 (#32162)

* fix command order

* add integration test

* test cleanup

* remove redudant commands

mattclay committed:
Update ansible-test sanity command. (#31958)

* Use correct pip version in ansible-test.
* Add git fallback for validate-modules.
* Run sanity tests in a docker container.
* Use correct python version for sanity tests.
* Pin docker completion images and add default.
* Split pylint execution into multiple contexts.
* Only test .py files in use-argspec-type-path test.
* Accept identical python interpeter name or binary.
* Switch cloud tests to default container.
* Remove unused extras from pip install.
* Filter out empty pip commands.
* Don't force running of pip list.
* Support delegation for windows and network tests.
* Fix ansible-test python version usage.
* Fix ansible-test python version skipping.
* Use absolute path for log in ansible-test.
* Run vyos_command test on python 3.
* Fix windows/network instance persistence.
* Add `test/cache` dir to classification.
* Enable more python versions for network tests.
* Fix cs_router test.

rcarrillocruz committed:
Remove duplex option on ios_interface (#32178)

Turns out on devices where only Gigabit is available (like our
csr nodes in CI), duplex is not configurable, thus the test fail.
Removing, since we are testing modification of other properties like
description and mtu and have good coverage.

rcarrillocruz committed:
Use sane mtu values on ios_interface tests (#32180)

Anything lower than 1500 won't work on CSR nodes.

igor-pinchuk committed:
Update hashi_vault.py

Add ability to lookup entire objects in HashiCorp Vault.
When used with auth_method allow skipping SSL verify.

rcarrillocruz committed:
Remove bogus interface description (#32183)

The interface has already that description from a task earlier, thus that change
is not introduced susbequently and we don't have to assert for it.

dgilbert82 committed:
No need to set a default tenancy and it prevents the creation of spot lcs (#31270)

The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_82dk2ynr/ansible_module_ec2_lc.py", line 317, in create_launch_config
    connection.create_launch_configuration(**launch_config)
  File "/usr/lib/python3.6/site-packages/botocore/client.py", line 312, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/lib/python3.6/site-packages/botocore/client.py", line 601, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ValidationError) when calling the CreateLaunchConfiguration operation: Placement tenancy is not supported for spot instances.

fatal: [localhost]: FAILED! => {
    "changed": false, 
    "error": {
        "code": "ValidationError", 
        "message": "Placement tenancy is not supported for spot instances.", 
        "type": "Sender"
    }, 
    "failed": true,

prasadkatti committed:
Document return values of ec2_eip (#32018)

- added RETURN value for ec2_eip
- removed note about what is returned
- removed a duplicate example

prasadkatti committed:
[ec2_vpc_subnet_facts] Add subnet_id alias for subnet_ids options (#32023)

prasadkatti committed:
[ec2-ami] make examples less verbose; fix default values in docs (#32008)

* make ec2-ami examples less verbose

* Fix default values in docs to be the actual default values

Fix default values for `architecture`, `virtualization_type` and
`wait_timeout` in docs to be the actual default values.

* Added note about examples not containing auth details

prasadkatti committed:
[ec2] Move note about state=restarted (#31961)

ryansb committed:
[cloud] Bugfix for aws_s3 empty directory creation (#32169)

* [cloud] Bugfix for aws_s3 empty directory creation

* Update integration tests

prasadkatti committed:
[cloud][docs] Add a cloudformation_facts example (#31964)

machacekondra committed:
ovirt_hosts: Don't fail upgrade when NON_RESPONSIVE state (#32123)

remixtj committed:
Use latest available template (#31780)

* Use latest available template

Documentation states:
template_version: version number of the template to be used for VM. By default the latest available version of the template is used.

This was not true because if parameter was not specified, template[0] is choosen, without checking if is the latest. Now, sorting + selecting the latest selects the one with the latest version number.

* Sort in reverse order, style cleanup

Applied fixes from comment

tedder committed:
aws_kms: handle updated policy format+cleanup (#30728)

* aws_kms: handle updated policy format+cleanup

- create slightly updated policy in that handles lists instead of a single string; the previous version's policy was being rejected if the key was new enough to have the updated base policy.
- removed `dry_run` conditionals, not committing the policy anyhow.
- return the policy in the return data. Leaving undocumented for now.
- update exception handling: don't rethrow in `do_grant`, don't pass anything to `format_exc`.

* whitespace/indent fail

* fix list-plus-brackets

* str and list fixes for ryansb

* port changes from #31667 over, better listification

Commit:
Integration Tests only: add static route, snmp_user, snapshot and hsrp it cases (#28933)

* add static route and hsrp it cases

* add snmp_user and snapshot it

* password strength correction

* fix typo

* add compare snapshot as it is fixed in the code now

* skip snapshot tests for titanium

* titanium and nxapi conditions added

samdoran committed:
Enable ECHO in prompt module (#32083)

* Enable ECHO in prompt module

Fixes #14160

* Set flags to make it possible to edit echoed input as well as hide control charcters

Only do this if a time limit is not set.

* Consolidate settings

mmercer-apixio committed:
Cast target port to an int in elb_target_group. Fixes #32098 (#32202)

samdoran committed:
Return all elements in a more robust way

If a trailing ':' is set or not, always return all secrets from a path. This was a bug mostly fixed by this PR.

Update examples.

tedder committed:
Exclude stack policy when running in check mode. (#31940)

Fixes #31931

s-hertel committed:
cloudformation_facts: don't fail on nonexistent stack - fixes #23419 (#23758)

* Allow cloudformation_facts to exit gracefully if stack does not exist

make cloudformation_facts pep8

remove from legacy files

remove unnecessary if statement

Allow cloudformation_facts to exit gracefully if stack does not exist version 2

fix documentation errors

add an example for a hard-fail if a stack doesn't exist

* Remove extra whitespace

* Use the .response attribute since .message isn't present with Python 3

* Don't fail if no stack name is provided and no stacks exist.

CXCV committed:
Enable TLS1.1 and TLS1.2 for win_package (#32184)

prasadkatti committed:
[ec2_vpc_nacl_id] Add nacl_id alias for nacl_ids option (#32223)

prasadkatti committed:
Route53 return values (#31962)

rcarrillocruz committed:
Handle ip name-server lines containing multiple nameservers (#32235)

In CSR, multiple nameservers are defined in one line, whereas on IOS
it's on multiple ones.
This change handles both.

trishnaguha committed:
Add vyos_vlan DI module (#32072)

* Add vyos_vlan DI module

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* Add integration tests

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* Improve logic and add more test

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* update boilerplate

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

rcarrillocruz committed:
Remove ip nameservers on ios_system/set_name_servers teardown (#32239)

Not sure why lookup source-interface, the only thing tested on that
file is adding/removing name servers, no lookup is set.

kedarkekan committed:
- Fix logging module issue where facility is being deleted along with host (#32234)

- Remove default facility, as facility is not standard on remote syslog server

kedarkekan committed:
Fixes ios_logging unit test (#32240)

rcarrillocruz committed:
Disable auto neg on test interfaces on IOS-XE (#32241)

As by default they are set on auto, test setting the speed on them
fail, we need to disable it upfront.

rcarrillocruz committed:
Enable authorize on disabling auto neg task (#32243)

skamithi committed:
Fixes #31090. In network parse_cli filter plugin, this change moves the creation of a (#31092)

new match(block). It previously only occurred if a new match occurred,
but suggest it should occur when an end of a block match is found.

ogenstad committed:
Remove asa_template action plugin (#32233)

Since the asa_template module has been deprecated this action plugin
is no longer needed.

prasadkatti committed:
[ec2_ami_copy] Add return documentation (#32012)

rcarrillocruz committed:
Change MTU values on ios_interface tests (#32251)

If setting 1500, we don't have 1500 in config as that's default value.
That causes issues on asserts.

s-hertel committed:
lambda: fix undefined variable and initially define account_id as None (#32249)

rcarrillocruz committed:
Fix lookup source tests on ios_system (#32254)

In IOS-XE, you need to pass an interface to lookup-source, otherwise
it fails with bad syntax.

rcarrillocruz committed:
Fix rest of ios_interface issues (#32257)

gregdek committed:
Update CONTRIBUTING.md to point to the right stuff (#32258)

We reorged these pages and these links haven't caught up, fixing

gregdek committed:
Moving guidelines to the official docs (#32260)

Redirecting contributors to the updated docs.

rcarrillocruz committed:
Fix eos_user tests (#32261)

bcoca committed:
added debug info when skipping undef

AlanCoding committed:
set the zip_safe flag to False (#32194)

The references to __file__ within code base make the zip_safe
method of installing not possible.

gregdek committed:
Update index.rst to point to new community guide (#32263)

We made this change a couple of months ago, deprecating community.html in favor of community/index.html and its full guide, but the change was never made here in the index.

gregdek committed:
Deprecate old community guide (#32266)

For anywhere we still have links, we want to deprecate this page and point them to the right new page.

bcoca committed:
ensure we always have a basedir

caphrim007 committed:
Adds the bigip_ssl_key module (#32270)

This module's purpose is to specifically manage the ssl keys. It
is essentially the key component of the bigip_ssl_certificate module.
The modules were separated and the key portion deprecated from
bigip_ssl_certificate in favor of this module.

bcoca committed:
more info on yaml errors

print yaml's 'problem'

trishnaguha committed:
testfix: IP address based on dci (#32279)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

edsantiago committed:
(minor) fix broken link, awkward phrasing (#32085)

* (minor) fix broken link, awkward phrasing

Simple transposition error was resulting in a link not
being properly htmlified.

Also clean up redundant 'this' and trailing whitespace.

Signed-off-by: Ed Santiago <santiago@redhat.com>

* Edits

caphrim007 committed:
Adds bigip_asm_policy module (#32281)

This module can be used to import asm policies from file or existing
template. Supported file types are xml, compact xml, and binary

bcoca committed:
minor format change

irsl committed:
become-user example was incorrect

Example was incorrect:
--become-user doesn't imply --become (http://docs.ansible.com/ansible/latest/become.html)

abadger committed:
Revert "Removed a force conditional (#28851)" (#32282)

This reverts commit 07acc579db839170122fc66505a886ef023d5f4f.

On closer examination of this code, the conditional that had force in it
was not a parent of this one.  So handling of force is needed i both
branches.

See the recent comments on #23391

paneu committed:
ios_system: Fix typo in unit test (#32284)

aniketkhisti committed:
PEP8 compliancy for os_server_volume module (#32306)

trondhindenes committed:
Added info to win_service docs (#32320)

* Added info to win_service docs

* minor typo fix

willthames committed:
Amazon kms_facts module (#26733)

* Amazon kms_facts module

Facts module for Amazon's Key Management Service

* kms_facts provide aliases

Return aliases for keys
Provide `alias` as a filter
Cope when tags can't be listed
Ensure everything is properly snake cased

* Rename kms_facts to aws_kms_facts

There may be conflicting KMS modules for other providers otherwise.

* Fix documentation, add aliases cache

Aliases are called many times, so add a cache

* Reduce amount of info on deleted keys

Getting info on a key is costly (2s) per key, so reduce
info on deleted keys.

* Add policy information to facts

* aws_kms_facts version update

Fix ridiculously long RETURN line

* Remove dangerous-default-value from aws_kms_facts

dagwieers committed:
Fix typos and docs

dagwieers committed:
archive: PEP8 compliancy, pylint and docs (#30842)

This PR includes;
- PEP8 compliancy fixes
- pylint fixes
- Documentation updates

dagwieers committed:
assemble: PEP8 compliancy, pylint and docs (#30844)

This PR includes;
- PEP8 compliancy fixes
- pylint fixes
- Documentation updates

dagwieers committed:
blockinfile: PEP8 compliancy, pylint and docs (#30845)

This PR includes;
- PEP8 compliancy fixes
- pylint fixes
- Documentation updates

dagwieers committed:
synchronize: PEP8 compliancy, pylint and docs (#30858)

This PR includes;
- PEP8 compliancy fixes
- pylint fixes
- Documentation updates

dagwieers committed:
tempfile: PEP8 compliancy, pylint and docs (#30859)

This PR includes;
- PEP8 compliancy fixes
- pylint fixes
- Documentation updates

dagwieers committed:
acl: pylint fixes and docs (#30841)

This PR includes;
- pylint fixes
- Documentation updates

dagwieers committed:
copy: pylint and docs (#30847)

This PR includes;
- pylint fixes
- Documentation updates

dagwieers committed:
file: pylint fixes and docs (#30849)

This PR includes;
- pylint fixes
- Documentation updates

dagwieers committed:
lineinfile: pylint fixes and docs (#30854)

This PR includes;
- pylint fixes
- Documentation updates

dagwieers committed:
at: PEP8 compliancy and doc fixes (#30881)

This PR includes:
- PEP8 compliancy fixes
- Documentation changes

dagwieers committed:
getent: PEP8 compliancy and doc fixes (#30885)

This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

dagwieers committed:
hostname: PEP8 compliancy and doc fixes (#30886)

This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

dagwieers committed:
seboolean: PEP8 compliancy and doc fixes (#30888)

This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

dagwieers committed:
service: PEP8 compliancy and doc fixes (#30890)

This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

dagwieers committed:
systemd: PEP8 compliancy and doc fixes (#30892)

This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

dagwieers committed:
user: PEP8 compliancy and doc fixes (#30895)

This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

dagwieers committed:
subversion: PEP8 compliancy and doc fixes (#30909)

This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

wilvk committed:
EC2 AMI boto3 modifications (#28506)

Migrate ec2_ami module to boto3

jborean93 committed:
win_uri: fix elif to elseif (#32330)

paulquack committed:
Remove references to provider in documentation, update license (#32242)

Commit:
pamd: fix issue with trailing commas in module_arguments

dagwieers committed:
Sort the PEP8 legacy files (#32339)

dagwieers committed:
Fix typo

nilhase25 committed:
fix pep8 issue in consul_kv.py (#32310)

jsumners committed:
Add check for sv binary (#32250)

* Add check for sv binary

This commit adds a check for the sv binary. If the binary is not present
then the module will fail.

Resolves issue #32248

* Change sv check to suggestion by @bcoca

jctanner committed:
Change the ssh+sudo debug messages to vvv for easier troubleshooting in awx+tower

prasadkatti committed:
Added some more info about the aws_s3 module's overwrite option (#31819)

bcoca commented:
@gibbsoft this PR is against stable-2.2 ... it should be pointing at devel branch instead

once you rebase you'll also note that docs are now inside the plugin itself

garbled1 committed:
Add the ability to rename a snapshot, or change it's description. (#31441)

Also add tests for vmware_guest_snapshot, but disable them due to
vcsim not fully supporting such operations yet.

Implement changes suggested in review.  Also fix same in remove_or_revert_snapshot() for consistency

nilhase25 committed:
Pep8 aos login.py (#32312)

* fix pep8 issue in consul_kv.py

* fix pep8 issue in aos_login.py

bcoca committed:
remove misleading group vars as they are flat (#32276)

* remove misleading group vars as they are flat

* fixed vars display

josephtate committed:
ec2_lc - cast volume_size for an EBS volume to an int (#32291)

bcoca committed:
Inventory and varmanager updates (#32054)

* better parsing erros

* fix inventorydata serialization

* dicts fix

Commit:
Fix error when there are no docker volumes present yet (#28145)

* Fix error when there are no docker volumes present yet

* fixes suggested by Shippable

sivel committed:
Fix patching to epel package

prasadkatti committed:
[aws] Document boto3 requirement (#32329)

paneu committed:
ios_logging: Fix some smaller issues, add unit test (#32321)

* ios_logging: Fix typo in documentation

* ios_logging: Fix traceback when setting buffered destination without size

When the size parameter is not configured while configuring the buffered
destination, a traceback occurs due to the fact that validate_size expects the
parameter to be an int. Explicitely converting value to int makes the
check work for every case.

* ios_logging: Update size parameter documentation

Update the documentation of the size paramter to reflect the current behaviour
of setting a default of 4096 for the buffered dest.

* ios_logging: Add unit test

Add unit test for ios_logging testing the behaviour clarified in the previous
commits.

* ios_logging: Fix python 2.6 compliance

ganeshrn committed:
Fix platform agnostic network integration test cases (#32378)

* Fix platform agnostic network test cases

* Fix vyos_interface CI failure

sbulage committed:
google: PEP8 compliancy fixes (#32309)

This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

Signed-off-by: Satyajit Bulage <sbulage@redhat.com>

mkrizek committed:
git: fix archive when update is set to no (#31829)

doubleplush committed:
add support to vmware_guest for template => vm conversion (#31607)

* add support to vmware_guest for template => vm conversion

While the vmware_guest currently supports conversion of VMs to templates
using the is_template argument, it does not support the inverse:
converting templates back into VMs.  This change adds that
functionality.

When converting a template back into a VM, the extra config option
"uuid.action" is also set so that VMware will automatically create a new
UUID for the converted VM.  If the "uuid.action" setting is already
configured, it will not be modified.  Setting this prevents an
interactive question from being raised when attempting to boot the VM.

* Add integration tests for vmware_guest is_template

* Add additional idempotency test for vmware_guest is_template

willthames committed:
Allow health_check_port to be a string (#32228)

Setting health_check_port to 'traffic-port' allows the health
check to use the target's traffic port.

gibbsoft committed:
Fixes #32346 Add default to credstash lookup

gibbsoft commented:
Closing.  Will open another against `devel` branch.
